src/openforms/submissions/models/submission_files.py

```diff
@@ -199,6 +199,41 @@ def informatieobjecttype(self) -> Optional[str]:
                 ):
                     return iotype
 
+    @property
+    def bronorganisatie(self) -> str | None:
+        """
+        Get the bronorganisatie for this attachment from the configuration
+        """
+        formdef_config = self.submission_step.form_step.iter_components()
+        for component in formdef_config:
+            if component["key"] == self.form_key:
+                if bron := glom(component, "registration.bronorganisatie", default=""):
+                    return bron
+
+    @property
+    def vertrouwelijkheidaanduiding(self) -> str | None:
+        """
+        Get the vertrouwelijkheidaanduiding for this attachment from the configuration
+        """
+        formdef_config = self.submission_step.form_step.iter_components()
+        for component in formdef_config:
+            if component["key"] == self.form_key:
+                if vertrouwelik := glom(
+                    component, "registration.vertrouwelijkheidaanduiding", default=""
+                ):
+                    return vertrouwelik
+
+    @property
+    def titel(self) -> str | None:
+        """
+        Get the title for this attachment from the configuration
+        """
+        formdef_config = self.submission_step.form_step.iter_components()
+        for component in formdef_config:
+            if component["key"] == self.form_key:
+                if titel := glom(component, "registration.titel", default=""):
+                    return titel
```

Comment:
this repetition around `iter_components` is not great for performance, especially because the same processing is done multiple times if all these attributes are accessed (which they are).

Please use the caching datastructure https://github.com/open-formulieren/open-forms/blob/master/src/openforms/forms/models/form_definition.py#L154